langserver/handlers/cancel_request.go

```diff
@@ -2,13 +2,29 @@ package handlers
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/creachadair/jrpc2"
-	lsp "github.com/sourcegraph/go-lsp"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
 )
 
 func CancelRequest(ctx context.Context, params lsp.CancelParams) error {
-	id := params.ID.String()
+	id, err := decodeRequestID(params.ID)
+	if err != nil {
+		return err
+	}
+
 	jrpc2.CancelRequest(ctx, id)
 	return nil
 }
+
+func decodeRequestID(v interface{}) (string, error) {
+	if val, ok := v.(string); ok {
+		return val, nil
```

Comment:
`sourcegraph/go-lsp` actually quoted string IDs

https://github.com/sourcegraph/go-lsp/blob/219e11d77f5d414b4fe5ba7e532b277fca34c1b4/jsonrpc2.go#L25

but I'm considering that a bug which we never encountered because of how rarely strings are used by language clients (most use numbers, which in case of `go-lsp` were interpreted as `uint64` and in gopls' as `float64`). Either way, if the client expects the server to quote the ID I would consider that a bug on the client side too, because from what I can tell the spec doesn't mention anything about quoting IDs:
https://microsoft.github.io/language-server-protocol/specifications/specification-current/#requestMessage